src/App.tsx

```diff
@@ -1,14 +1,30 @@
 import { AppHeader } from "./components/AppHeader";
 import { AppLayout } from "./components/AppLayout";
-import {Card} from "./components/Card";
+import {NewCard} from "./components/NewCard";
+import React, { useState } from "react";
 
 import "./App.css";
-import React from "react";
 
 function App() {
+  const [editMode, setEditMode] = useState<boolean>(true);
+  const [cardList, setCardList] = useState<JSX.Element[]>([]); 
```

Comment:
**issue:** Keeping array of elements is not react way of doing it. You should handle data structure here, as array of objects, then you should map this data as react components. You will find a good example in react docs of how can you render lists: https://react.dev/learn/rendering-lists